src/plugins/cuda_gds/gds_backend.cpp

```diff
@@ -159,7 +202,19 @@ nixl_status_t nixlGdsEngine::deregisterMem (nixlBackendMD* meta)
     nixlGdsMetadata *md = (nixlGdsMetadata *)meta;
     if (md->type == FILE_SEG) {
         gds_utils->deregisterFileHandle(md->handle);
-	gds_file_map.erase(md->handle.fd);
+
+        // Find and erase the entry using devId
+        for (auto it = gds_file_map.begin(); it != gds_file_map.end(); ++it) {
+            if (it->second.fd == md->handle.fd) {
+                gds_file_map.erase(it);
+                break;
+            }
+        }
```

Comment:
It seems that you can use `md->handle.fd` to find the entity.
```suggestion
        const auto& it = gds_file_map.find(md->handle.fd);
        if (it != gds_file_map.end()) {
            gds_file_map.erase(it);
        }
```